data/src/main/scala/cats/data/OneAnd.scala

```diff
@@ -1,32 +1,105 @@
 package cats
 package data
 
-final case class OneAnd[A, F[_]](head: A, tail: F[A])
+/**
+ * A data type which represents a single element (head) and some other
+ * structure (tail). As we have done in package.scala, this can be
+ * used to represent a List which is guaranteed to not be empty:
+ * {{{
+ * type NonEmptyList[A] = OneAnd[A, List]
+ * }}}
+ */
+final case class OneAnd[A, F[_]](head: A, tail: F[A]) {
+
+  /**
+   * remove elements not matching the predicate
+   */
+  def filter(f: A => Boolean)(implicit monad: MonadCombine[F]): F[A] = {
+    val rest = monad.filter(tail)(f)
+    if(f(head))
+      monad.combine(monad.pure(head), rest)
+    else
+      rest
+  }
+
+  /**
+   * Append another OneAnd to this
+   */
+  def combine(other: OneAnd[A,F])(implicit monad: MonadCombine[F]): OneAnd[A,F] =
+    OneAnd(head, monad.combine(tail, monad.combine(monad.pure(other.head), other.tail)))
+
+  /**
+   * find the first element matching the predicate, if one exists
+   */
+  def find(f: A => Boolean)(implicit foldable: Foldable[F]): Option[A] =
+    if(f(head)) Some(head) else foldable.find(tail)(f)
+
+  /**
+   * Left associative fold on the structure using f
+   */
+  def foldLeft[B](b: B)(f: (B, A) => B)(implicit foldable: Foldable[F]): B =
+    foldable.foldLeft(tail, f(b, head))(f)
+
+  /**
+   * Right associative fold on the structure using f
+   */
+  def foldRight[B](b: B)(f: (A,B) => B)(implicit foldable: Foldable[F]): B =
+    f(head, foldable.foldRight(tail, b)(f))
+}
 
 trait OneAndInstances {
 
   implicit def oneAndShow[A, F[_]](implicit showHead: Show[A], showTail: Show[F[A]]): Show[OneAnd[A, F]] =
     Show.show[OneAnd[A, F]](x => s"OneAnd(${showHead.show(x.head)}, ${showTail.show(x.tail)})")
 
-  implicit def instance[F[_]](implicit monoid: MonoidK[F], functor: Functor[F], monad: Monad[F]): Comonad[OneAnd[?, F]] with Monad[OneAnd[?, F]] = new Comonad[OneAnd[?, F]] with Monad[OneAnd[?, F]] {
+  implicit def oneAndFunctor[F[_]](F: Functor[F]) = new Functor[OneAnd[?,F]] {
```

Comment:
Good catch. If I recall, someone recently did a sweep and added them to all of the places that they were missing.